src/System.Private.ServiceModel/tests/Benchmarks/PerfUsingCoreWCFService/WCFCorePerfClient/Program.cs

```diff
@@ -0,0 +1,200 @@
+        {
+            Console.WriteLine("WCFCorePerf Client.");
+
+            Program test = new Program();
+
+            if (test.ProcessRunOptions(args))
+            {
+                var startTime = DateTime.Now;
+                int request = 0;
+
+                BenchmarksEventSource.Register("wcfcoreperf/channelopen", Operations.Max, Operations.Max, "Channel Open Time (ms)", "Time to Open Channel in ms", "n0");
+                BenchmarksEventSource.Register("wcfcoreperf/firstrequest", Operations.Max, Operations.Max, "First Request (ms)", "Time to first request in ms", "n0");
+                BenchmarksEventSource.Register("wcfcoreperf/requests", Operations.Max, Operations.Sum, "Requests (" + test._paramPerfMeasurementDuration.TotalMilliseconds + " ms)", "Total number of requests", "n0");
+                BenchmarksEventSource.Register("wcfcoreperf/rps/max", Operations.Max, Operations.Sum, "Requests/sec (max)", "Max requests per second", "n0");
+
+                switch (test._paramBinding)
+                {
+                    case TestBinding.BasicHttp:
+                        BasicHttpBinding binding = new BasicHttpBinding();
+                        switch (test._paramTransferMode.ToLower())
+                        {
+                            case "buffered":
+                                binding.TransferMode = TransferMode.Buffered;
+                                break;
+                            case "streamed":
+                                binding.TransferMode = TransferMode.Streamed;
+                                break;
+                            case "streamedrequest":
+                                binding.TransferMode = TransferMode.StreamedRequest;
+                                break;
+                            case "streamedresponse":
+                                binding.TransferMode = TransferMode.StreamedResponse;
+                                break;
+                            default:
+                                break;
+                        }
+
+                        Console.WriteLine($"Testing TransferMode: {binding.TransferMode}");
+                        ChannelFactory<ISayHello> factory = new ChannelFactory<ISayHello>(binding, new EndpointAddress(test._paramServiceUrl));
+                        var stopwatchChannelOpen = new Stopwatch();
+                        stopwatchChannelOpen.Start();
+                        factory.Open();
+                        BenchmarksEventSource.Measure("wcfcoreperf/channelopen", stopwatchChannelOpen.ElapsedMilliseconds);
+
+                        var client = factory.CreateChannel();
+                        var stopwatchFirstReq = new Stopwatch();
+                        stopwatchFirstReq.Start();
+                        var result = client.HelloAsync("helloworld").Result;
+                        BenchmarksEventSource.Measure("wcfcoreperf/firstrequest", stopwatchFirstReq.ElapsedMilliseconds);
```

Comment:
If we're going to be using synchronous code, we should have the contract have be a synchronous method. .NET allows you to make the Main method async now so this should probably be awaiting the async method. We say async is the way people should use it, but we should measure sync too.